torch/autograd/_functions/tensor.py

```diff
@@ -557,21 +556,19 @@ def backward(ctx, grad_output):
 
 class Cumsum(Function):
 
-    def __init__(self, dim):
-        super(Cumsum, self).__init__()
-        self.dim = dim
-
-    def forward(self, input):
-        return torch.cumsum(input, dim=self.dim)
+    @staticmethod
+    def forward(ctx, input, dim):
+        ctx.dim = dim
+        return torch.cumsum(input, dim=ctx.dim)
 
-    def backward(self, grad_output):
-        grad_input = torch.cumsum(-grad_output, dim=self.dim)
+    @staticmethod
+    def backward(ctx, grad_output):
+        grad_input = torch.cumsum(-grad_output, dim=ctx.dim)
 
-        end_idx = grad_input.size(self.dim) - 1
-        grad_sum = grad_input.narrow(self.dim, end_idx, 1)
-        grad_input -= grad_sum.expand_as(grad_input)
-        grad_input += grad_output
-        return grad_input
+        end_idx = grad_input.size(ctx.dim) - 1
+        grad_sum = grad_input.narrow(ctx.dim, end_idx, 1)
+        grad_input = (grad_input - grad_sum.expand_as(grad_input)) + grad_output
```

Comment:
```
grad_input = (grad_input - grad_sum.expand_as(grad_input))
grad_input += grad_output
```
seems to work.  Make the first line inplace and running through gradgrad check gives me:
```
RuntimeError: in-place operations can be only used on variables that don't share storage with any other variables, but detected that there are 3 objects sharing it
```